oak_runtime/src/node/http/tests.rs

```diff
@@ -20,17 +20,71 @@ use oak_abi::{
     label::Label,
     proto::oak::application::{
         node_configuration::ConfigType, ApplicationConfiguration, HttpServerConfiguration,
-        NodeConfiguration,
+        NodeConfiguration, WebAssemblyConfiguration,
     },
 };
-use std::{fs, option::Option, thread::JoinHandle};
+use std::fs;
 
 static LOCAL_CA: &str = "../examples/certs/local/ca.pem";
 static GCP_CA: &str = "../examples/certs/gcp/ca.pem";
 
+struct OakNodeForTest {}
+
+impl crate::node::Node for OakNodeForTest {
```

Comment:
I assume we will want to generalize this so that it can be reused in other places, we only need to provide the body of the node for each particular test, but hopefully not an entire new factory and node type every time.

Reply:
With the new approach, I am not sure if generalising it has much value. I think each test module can define a set of test Oak nodes, and provide the implementation for the `run` method.